File: app_java/src/main/java/com/dwp/mvvmbasicsample/TaskViewModel.java
package com.dwp.mvvmbasicsample;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import com.dwp.mvvmbasicsample.room.Task;

import java.util.List;

/**
 * Created by dwp on 2020-11-03.
 */
public class TaskViewModel extends AndroidViewModel {

    TaskRepository repository;
    LiveData<List<Task>> allTasks;

    public TaskViewModel(@NonNull Application application) {
        super(application);

        repository = new TaskRepository(application);
        allTasks = repository.getAllTasks();
    }

    public void insert(Task task) {
        repository.insert(task);
    }

    public void update(Task task) {
        repository.update(task);
    }

    public void delete(Task task) {
        repository.delete(task);
    }

    public void deleteAllTasks() {
        repository.deleteAllTasks();
    }

    public LiveData<List<Task>> getAllTasks() {
        return allTasks;
    }
}

File: app_kotlin/src/main/java/com/dwp/mvvmbasicsample/TaskViewModel.kt
package com.dwp.mvvmbasicsample

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import com.dwp.mvvmbasicsample.room.Task

/**
 * Created by dwp on 2020-11-02.
 */

class TaskViewModel(application: Application) : AndroidViewModel(application) {
    private val repository = TaskRepository(application)
    private val allTasks = repository.getAllTasks()

    fun insert(task: Task) {
        repository.insert(task)
    }

    fun update(task: Task) {
        repository.update(task)
    }

    fun delete(task: Task) {
        repository.delete(task)
    }

    fun deleteAllTasks() {
        repository.deleteAllTasks()
    }

    fun getAllTasks(): LiveData<MutableList<Task>> {
        return allTasks;
    }
}